orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/PipelineTemplateService.java

```diff
@@ -87,9 +89,8 @@ public Execution retrievePipelineOrNewestExecution(@Nullable String executionId,
       ExecutionRepository.ExecutionCriteria criteria = new ExecutionRepository.ExecutionCriteria().setLimit(1);
       try {
         return executionRepository.retrievePipelinesForPipelineConfigId(pipelineConfigId, criteria)
```

Comment:
any issue if the underlying result was empty ... ?

Reply:
The code here was already designed around the fact that it'll handle a `NoSuchElementException`. I could've further refactored this to not behave in this way, but didn't want unrelated changes to blur things up.